### PR TITLE
Don't consider intermittent failures which were already intermittent in a close parent as regressions

### LINKED ISSUE #1: An intermittent should not be considered a regression if it was already intermittent in one of its parents
It's not perfectly clear, as maybe a new intermittent was introduced on the current push, but it might be better not to consider intermittently failing tasks that were already intermittently failing in one of the parents.

### PR DESCRIPTION
**Pull request by marco-c**
Fixes #1

### CODE REVIEW

File: mozci/task.py [hunk 1]
```diff
@@ -362,6 +362,12 @@ def configuration(self):
 # Don't perform type checking because of https://github.com/python/mypy/issues/5374.
 @dataclass  # type: ignore
 class RunnableSummary(ABC):
+    @property
+    def is_intermittent(self):
+        return self.status == Status.INTERMITTENT or any(
+            c == "intermittent" for c, n in self.classifications
```
Comment (ahal) [inline mozci/task.py:368]: There's also an `autoclassified intermittent` classification.. not sure how much it actually gets used (if ever), but might as well support it I think.

File: mozci/push.py [hunk 2]
```diff
@@ -894,20 +902,40 @@ def get_regressions(self, runnable_type: str) -> Dict[str, int]:
...
                     if name in runnable_summaries:
+                        found_in_parent = True
                         summary = runnable_summaries[name]
-                        if summary.status != Status.PASS and all(
-                            c != "intermittent" for c, n in summary.classifications
-                        ):
+
+                        # If the failure is not intermittent...
+                        if not failure_summary.is_intermittent:
+                            # ...and it failed permanently in the first parent where it run, it is a
```
Comment (ahal) [inline mozci/push.py:916]: nit: s/run/ran

**Review by ahal — APPROVED** [review]

File: mozci/task.py (hunk 1 above)
Comment (marco-c) [reply mozci/task.py:368]: Is it set automatically or is it still chosen by sheriffs?

Comment (marco-c) [reply mozci/task.py:368]: (I'm wondering if we can trust it as we can trust `intermittent`)

Comment (marco-c) [reply mozci/task.py:368]: I've been told sheriffs don't use it anyway, so I'll not add it. Perhaps we can remove it from Treeherder options? CC @camd

**Pull request merged**